src/components/ListItem.tsx

```diff
@@ -1,8 +1,70 @@
-import * as api from '../api';
-import './ListItem.css';
+import "./ListItem.css";
+import { updateItem, ListItem } from "../api";
+import { useEffect, useState } from "react";
+import toast from "react-hot-toast";
+import { is24HoursLater } from "../utils";
 
-type Props = Pick<api.ListItem, 'name'>;
+interface Props {
+	item: ListItem;
+	listPath: string | null;
+}
+
+export function ListItemCheckBox({ item, listPath }: Props) {
+	const [checked, setChecked] = useState(false);
+
+	const purchaseDate = item.dateLastPurchased
+		? item.dateLastPurchased.toDate()
+		: item.dateLastPurchased;
+
+	useEffect(() => {
+		if (purchaseDate) {
+			setChecked(!is24HoursLater(purchaseDate));
+		} else {
+			setChecked(false);
+		}
+	}, [item.dateLastPurchased]);
+
```

Comment:
Suggestion: The useEffect hook re-calculates the checked state every time item.dateLastPurchased changes. This is fine, but it could be simplified slightly:
```
useEffect(() => {
  setChecked(purchaseDate ? !is24HoursLater(purchaseDate) : false);
}, [purchaseDate]);
```
This version uses `purchaseDate` directly as a dependency in the useEffect instead of `item.dateLastPurchased`, and also reduces the cognitive load inside the hook.

Reply:
Since javascript date objects are mutable I had suggested that the depenency array use the "source of truth" `item.dateLastPurchased`, it would make sure any change in the `useEffect` only happen if there is a change in the `item.dateLastPurchased` keeping it reliable and without the concern of if `purchaseDate` got mutated at some point.

I could be not thinking about the direction quite right. Not sure if @tannaurus, @alex-andria or @adidalal would be able to explain it a bit more.

Reply:
Frankly it's been too longer since I've written JS to know confidentially that will work, but I think it will 🙂

Reply:
Also, tagged unions are by far the best feature in Typescript. I live and breath them for component state modeling

Reply:
I just realized I used `.await` 🤦 you'll have to translate that for me to valid JS. Whoops!

Reply:
@tannaurus is this for the useEffect for the handling of the unchecking at 24 hours? Like is having the no use state would handle this without the useEffect?

Reply:
I think it would be cool to implement tagged unions in this way since we're going the TS route! I do agree that using the `dateLastPurchased` would be the most accurate way to pull that info from firebase, which is how I've done it in the past. 

If you decide to go the state checking route, which I think I've found common with regular React apps for TCL in the past, it would be a fun exercise to use tagged unions to differentiate local state and firebase state like Tanner said for our TS app and then on firebase query have the local override (update) the storage. 

We can maybe do this during another mini office hours together and make it a refactor spike card for the next sprint. I'm a bit rusty on my TS but open to exploring this soon! We can always add/enhance our modal rendering later too as a UI card later in the cohort, but at the least have our error catching logic in place in the meantime.

Reply:
Hmm!!! I really appreciate this feedback and convo.  I can now see since we already update item.dateLastPurchased anyways within our updateItem call, to save the cognitive load as Falak was mentioning, figuring out how to update the checkbox state within that updateItem call could be really efficient!!